code/game/gamemodes/vampire/vampire.dm

```diff
@@ -0,0 +1,126 @@
+/datum/game_mode
+	var/list/datum/mind/vampires = list()
+
+
+/mob/proc/make_mob_into_vampire() //For easy calling on mobs
+	make_vampire(src)
+
+
+/proc/is_vampire(var/mob/living/M)
+	return istype(M) && M.mind && ticker && ticker.mode && (M.mind in ticker.mode.vampires)
+
+
+/proc/change_vampire_state(var/mob/living/M, var/remove)
+	if(!remove)
+		ticker.mode.vampires.Add(M)
+	else
+		ticker.mode.vampires.Remove(M)
+
+
+/datum/vampire //This datum handles stuff specific to the vampire mini-antag
+	var/sucked_blood //The amount of blood a vampire has stolen
+	var/obj/structure/closet/coffin/marked_coffin //The coffin the vampire has marked as home
+
+
+/proc/make_vampire(var/mob/living/carbon/human/V, var/silent = 0)
+	if(!V || !istype(V) || !V.mind || !V.client) //Only works on humans that aren't braindead
+		return
+	var/datum/mind/M = V.mind
+	if(!silent)
+		V << "<span class='userdanger'>You are a vampire!</span>"
+		V << "<b>You are a vampire, a supernatural, nearly-immortal creature nourished by blood.</b>"
+		V << "<b>You cannot die to many normal means. Blunt trauma, fire, toxins, and oxygen deprivation will never kill you. In addition, you will slowly regenerate most damage types while you have blood in \
+		your body.</b>"
+		V << "<b>However, you have some weaknesses, and chief among these is the sun. If caught in starlight, you will rapidly take damage and soon die. In addition, holy objects will harm you. Simply being \
+		in consecrated areas will drain your power, and holy water will light you ablaze.</b>"
+		V << "<b>Upon death, you will turn to ashes and bones. If you have marked a coffin, you will eventually revive there with no items.</b>"
+		V << "<b>You are nourished by blood. You will not regenerate it naturally, and must gain it from living and sapient humans. To do this, simply target the head and click on a human, preferably asleep, \
+		with an empty hand on harm intent. You will begin silently draining their blood, but never enough to kill them.</b>"
+		V << "<b>If you run out of blood, your immortality will fade and you will be vulnerable to all damage until you gain enough blood to sustain yourself once more.</b>"
+		V << "<b>You can be distinguished as vampire by health analyzers encountering errors while scanning you. Try to avoid this.</b>"
```

Comment:
There needs to be a way to detect them, and the actual detection tell is fairly subtle.